bd-logger/src/async_log_buffer.rs

```diff
@@ -490,23 +529,21 @@ impl<R: LogReplay + Send + 'static> AsyncLogBuffer<R> {
             anyhow::bail!("failed to send log processing completion message: {e:?}");
           }
         }
+
+        vec![]
       },
-      LoggingState::Initialized(initialized_logging_context) => {
-        if let Err(e) = self
-          .replayer
-          .replay_log(
-            log,
-            log_processing_completed_tx,
-            &mut initialized_logging_context.processing_pipeline,
-          )
-          .await
-        {
-          anyhow::bail!("failed to replay async log buffer log: {e}");
-        }
-      },
-    }
+      LoggingState::Initialized(initialized_logging_context) => self
+        .replayer
+        .replay_log(
+          log,
+          log_processing_completed_tx,
+          &mut initialized_logging_context.processing_pipeline,
+        )
+        .await
+        .map_err(|e| anyhow!("failed to replay async log buffer log: {e}"))?,
```

Comment:
since you're here this could probably be using `context()` instead

Reply:
OK let me do that.